Week 2/Solution.java

```diff
@@ -0,0 +1,41 @@
+import java.io.*;
+import java.util.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+public class Solution {
+    public int lengthOfLongestSubstring(String s) {
+        if(s==null || s.length() == 0) return 0;
+        int len = s.length();
+        int max = 1;
+        HashMap<Character,Integer> seen_pos = new HashMap<Character,Integer>();
```

Comment:
Great job - using HashMap (or something equivalent allowing for O(1) for update/retrieval) is the optimal solution for this problem!